subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java

```diff
@@ -34,7 +34,7 @@
 
 public class GradleVersion implements Comparable<GradleVersion> {
     public static final String URL = "http://www.gradle.org";
-    private static final Pattern VERSION_PATTERN = Pattern.compile("((\\d+)(\\.\\d+)+)(-(\\p{Alpha}+)-(\\d+[a-z]?))?(-(SNAPSHOT|\\d{14}([-+]\\d{4})?))?");
+    private static final Pattern VERSION_PATTERN = Pattern.compile("((\\d+)(\\.\\d+)+)(-(\\p{Alpha}+)-([0-9a-z]*))?(-(SNAPSHOT|\\d{14}([-+]\\d{4})?))?");
```

Comment:
Why are we using `*` instead of `+` for the second part? I think it should be `+`, if not something like `5.1-branchy--SNAPSHOT would be a valid version.

How about we allow upper case letters for the second part, too?

Reply:
Use `\w+`.